api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -558,6 +521,33 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
             .workspace(workspaceMapper.toApiWorkspace(savedWorkspace, toFcWorkspace)));
   }
 
+  // A retry period is needed because the permission to copy files into the cloned workspace is not granted transactionally
+  private void copyBlobWithRetries(String bucketName, Blob b, int retryPeriodSeconds) {
+    final int SLEEP_INTERVAL_SECONDS = 5;
+
+    try {
+      cloudStorageService.copyBlob(b.getBlobId(), BlobId.of(bucketName, b.getName()));
+    } catch (StorageException e) {
+      if (retryPeriodSeconds <= 0) {
+        throw e;
+      }
+
+      log.info(
+          "Service Account does not have access to bucket "
+              + bucketName
+              + ". Retrying for another "
+              + retryPeriodSeconds
+              + " seconds.");
+      try {
+        Thread.sleep(SLEEP_INTERVAL_SECONDS * 1000);
+      } catch (InterruptedException ex) {
+        e.printStackTrace();
```

Comment:
So here we're printing the exception, not logging an error or warning, and then just continuing?

I think we could use a simpler pattern here than this try/catch/try/throw logic.